x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/function/scalar/datetime/DateTimeFunction.java

```diff
@@ -22,147 +20,192 @@
 import org.joda.time.DateTime;
 import org.joda.time.DateTimeZone;
 
-import java.time.Instant;
-import java.time.ZoneId;
-import java.time.ZonedDateTime;
-import java.time.temporal.ChronoField;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Locale;
 import java.util.Objects;
 import java.util.TimeZone;
+import java.util.stream.Collectors;
 
 import static org.elasticsearch.xpack.sql.expression.function.scalar.script.ParamsBuilder.paramsBuilder;
-import static org.elasticsearch.xpack.sql.expression.function.scalar.script.ScriptTemplate.formatTemplate;
 
-public abstract class DateTimeFunction extends UnaryScalarFunction {
```

Comment:
DateTimeFunction is no longer a unary scalar function, it is now var args...

1= date
2=optional string holding timezoneid
3=optional string holding locale.

Note the right arg count is enforced based on the "type" of sql function in the ctor